backend/dataall/base/utils/iam_policy_utils.py

```diff
@@ -0,0 +1,95 @@
+from typing import List
```

Comment:
Can we maybe add a couple more comments on the double `while` loops to make easier to follow on what is going on / how we a determining the final policy statement set for each of the methods